### PR TITLE
final two hints

### LINKED ISSUE #3540: general problem issues
(by kevin sheng)

this is gonna be a big issue, imma gradually put down the list of problems that i feel need hints or those that are missing languages, contact me on discord if you feel a problem needs/doesn't need hints

# a whole-a** internal solution is missing
* [x] [photoshoot](https://usaco.guide/bronze/intro-greedy#problem-usaco-1227)

# hints

* [x] [sleepy cow herding](https://usaco.guide/problems/usaco-915-sleepy-cow-herding/solution) (wait why is this marked as easy)
* [x] [sleepy cow sorting](https://usaco.guide/problems/usaco-892-sleepy-cow-sorting/solution)
* [x] [milking order](https://usaco.guide/problems/usaco-832-milking-order/solution)
* [x] [taming the herd](https://usaco.guide/problems/usaco-809-taming-the-herd/solution)
* [x] [modern art](https://usaco.guide/problems/usaco-737-modern-art/solution)
* [x] [hoofball](https://usaco.guide/problems/usaco-808-hoofball/solution)
* [x] [spaced out](https://usaco.guide/problems/usaco-1088-spaced-out/solution)
* [x] [even more odd photos](https://usaco.guide/problems/usaco-1084-even-more-odd-photos/solution)
* [x] [race](https://usaco.guide/problems/usaco-989-race/solution)
* [x] [grass planting](https://usaco.guide/problems/usaco-894-grass-planting/solution)
* [x] [the great revegetation](https://usaco.guide/problems/usaco-916-the-great-revegetation/solution)
* [x] [milk factory](https://usaco.guide/problems/usaco-940-milk-factory/solution)
* [x] [swapity swap](https://usaco.guide/problems/usaco-1013-swapity-swap/solution)
* [x] [cow evolution](https://usaco.guide/problems/usaco-941-cow-evolution/solution)
# c++ missing

# java missing
* [x] [milking order](https://usaco.guide/problems/usaco-832-milking-order/solution)
* [x] [out of place](https://usaco.guide/problems/usaco-785-out-of-place/solution)

# python missing

### PR DESCRIPTION
**Pull request by SansPapyrus683**
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.

closes #3540 finally

### CODE REVIEW

**Review by ryanchou-dev — APPROVED** [review]
u don't know how amazing u are for sansifying all the code

File: solutions/bronze/usaco-941.mdx [hunk 1]
```diff
@@ -16,6 +16,13 @@ What's a defining trait about this bad set that you can check through code?
 
 </Spoiler>
 
+
+<Spoiler title="Answer to Hint">
+This would happen if we had a subpopulation with trait $A$, one with $B$, and one with $A$ and $B$.
+
+This doesn't work because this forces the cow with $A$ and the cow with $B$ to be in different subtrees, and so the cow with $A$ and $B$ would have to develop $A$ or $B$ randomly (depending on which subtree it stems from).
+</Spoiler>
+
```
Comment (SansPapyrus683) [inline solutions/bronze/usaco-941.mdx:25]:
```suggestion
```
this is the solution...

File: solutions/bronze/usaco-941.mdx [hunk 2]
```diff
@@ -5,15 +5,33 @@ title: Cow Evolution
...
-[Official Analysis (C++)](http://www.usaco.org/current/data/sol_evolution_bronze_open19.html)
-
 ## Video Solution
+
 <Youtube id="4YTijr0BbrA" />
 
+<Spoiler title="Hint">
+
+Try to construct a minimal set of cows that can't form a valid evolutionary tree.
+What's a defining trait about this bad set that you can check through code?
```
Comment (ryanchou-dev) [inline solutions/bronze/usaco-941.mdx:15]:
```suggestion
How can we generalize to different trees?
```

Comment (ryanchou-dev) [reply solutions/bronze/usaco-941.mdx:15]: i think telling the reader to check it in code implies that this minimal set is the only thing to take care of...

(would it be better to have the reader try to prove it themselves?)

File: solutions/bronze/usaco-1013.mdx [hunk 3]
```diff
@@ -116,66 +117,163 @@ public class swap {
...
+#include <numeric>
+#include <set>
+#include <vector>
+
+using std::cout;
+using std::endl;
+using std::vector;
+
+/** Reverses [start, end] in the given vector in-place. */
+template <typename T> void reverse_segment(vector<T> &vec, int start, int end) {
```
Comment (envyaims) [inline solutions/bronze/usaco-1013.mdx:160]: template isnt necessary but ok

**Review by envyaims — APPROVED** [review]

**Pull request merged**